src/components/themed/BuyCrypto.js

```diff
@@ -10,22 +10,20 @@ import { SPECIAL_CURRENCY_INFO } from '../../constants/WalletAndCurrencyConstant
 import s from '../../locales/strings.js'
 import { connect } from '../../types/reactRedux.js'
 import { Actions } from '../../types/routerTypes.js'
-import { getTokenId } from '../../util/CurrencyInfoHelpers.js'
 import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
 import { CurrencyIcon } from '../themed/CurrencyIcon.js'
 import { EdgeText } from './EdgeText.js'
 import { ButtonBox } from './ThemedButtons.js'
 
-const allowedTokenIds = Object.keys(SPECIAL_CURRENCY_INFO).filter(pluginId => !!SPECIAL_CURRENCY_INFO[pluginId].displayBuyCrypto)
+const allowedPluginIds = Object.keys(SPECIAL_CURRENCY_INFO).filter(pluginId => !!SPECIAL_CURRENCY_INFO[pluginId].displayBuyCrypto)
 
 type OwnProps = {
   // eslint-disable-next-line react/no-unused-prop-types
   walletId: string,
-  // eslint-disable-next-line react/no-unused-prop-types
-  currencyCode: string
+  tokenId?: string
```

Comment:
Pass the 'wallet' instead of a walletID, then remove the connector in the BuyCrypto and update BuyCrypto into a functional component